city_scrapers/spiders/fortx_Fort_Worth_Public_Meetings.py

```diff
@@ -0,0 +1,118 @@
+
+    meetings_url_payload = {
+        "LanguageCode": "en-US",
+        "Ids": ["8efac0b6-9ea3-402e-b7d9-e9e71a2a34a0"],
+        "StartDate": "",
+        "EndDate": "",
+    }
+
+    meeting_detail_url = (
+        "https://www.fortworthtexas.gov/ocapi/get/contentinfo"
+        "?calendarId={calendarId}&contentId={contentId}&language=en-US"
+        "&currentDateTime={currentDateTime}&mainContentId={mainContentId}"
+    )
+
+    def start_requests(self):
+        """
+        The meeting items for this organization are
+        fetched from two of its API endpoints. The
+        main API endpoint allows fetching meeting
+        items for the entirety of one year. The
+        spider is set to fetch all meetings for the
+        current year.
+        """
+        current_date = datetime.now().date()
+
+        start_of_year = datetime(current_date.year, 1, 1).date()
+        end_of_year = datetime(current_date.year, 12, 31).date()
+
+        self.meetings_url_payload["StartDate"] = str(start_of_year)
+        self.meetings_url_payload["EndDate"] = str(end_of_year)
+
+        yield scrapy.Request(
+            url=self.meetings_url,
+            method="POST",
+            body=json.dumps(self.meetings_url_payload),
+            headers={"Content-Type": "application/json"},
+            callback=self.parse,
+        )
+
+    def parse(self, response):
+        data = response.json()
+
+        items = []
+        for meeting in data["data"]:
+            items.extend(meeting["Items"])
+
+        for item in items:
+            date_obj = datetime.strptime(item["DateTime"], "%d/%m/%Y %I:%M:%S %p")
+            currentDateTime = date_obj.strftime("%d/%m/%Y%%20%I:%M:%S%%20%p")
+
+            meeting_detail_url = self.meeting_detail_url.format(
+                calendarId=item["CalendarId"],
+                contentId=item["Id"],
+                currentDateTime=currentDateTime,
+                mainContentId=item["MainContentId"],
+            )
+
+            yield scrapy.Request(
+                url=meeting_detail_url,
+                method="GET",
+                callback=self.parse_meeting,
+                cb_kwargs={"item": item},
+            )
+
+    def parse_meeting(self, response, item):
+        data = response.json()
+        meeting_data = data["data"]
+
+        meeting = Meeting(
+            title=meeting_data["Title"],
+            description=self._parse_description(meeting_data),
+            classification=CITY_COUNCIL,
+            start=dateparse(item["DateTime"]),
+            end=None,
+            all_day=False,
+            time_notes="Please check the meeting description for details on the start time",  # noqa
+            location=self._parse_location(meeting_data),
+            links=[],
+            source=response.url,
+        )
+
+        meeting["status"] = self._get_status(meeting)
+        meeting["id"] = self._get_id(meeting)
+
+        yield meeting
```

Comment:
Let's take a look at a sample meeting data scraped from the API: https://www.fortworthtexas.gov/ocapi/get/contentinfo?calendarId=8efac0b6-9ea3-402e-b7d9-e9e71a2a34a0&contentId=3ebb347a-3db1-425c-8669-a0a5d329273b&language=en-US&currentDateTime=14/12/2024%2011:20:00%20AM&mainContentId=3ebb347a-3db1-425c-8669-a0a5d329273b

+ Please use the key "Link" for `meeting.source`. We prefer to get a "website" URL for this value. API URL is the last resort. In this sample meeting:
  - it is https://www.fortworthtexas.gov/departments/parks/events/meetings-and-projects/prairie-dog-revised-master-meeting
  - NOT https://www.fortworthtexas.gov/ocapi/get/contentinfo?calendarId=8efac0b6-9ea3-402e-b7d9-e9e71a2a34a0&contentId=3ebb347a-3db1-425c-8669-a0a5d329273b&language=en-US&currentDateTime=14/12/2024%2011:20:00%20AM&mainContentId=3ebb347a-3db1-425c-8669-a0a5d329273b

+ Please refer to the key `IsCancelled` to parse status "CANCELLED" for this meeting if it is `True`, not just use the inherited method `meeting["status"] = self._get_status(meeting)`. We usually have a method `_parse_status` to do this job
+ There are some issues with method `_parse_location(self, item)`:
  - Please don't take "isCancelled" into account
  - The key "address" includes location's name. Please exclude "name" from "address". Or you can build the "address" key from the other keys: `address = f"{Street}, {Suburb}, {PostCode}"`. Expected parsed location would be
```json
{
      "name": "Eugene McCray Community Center",
      "address": "4932 Wilbarger St, Fort Worth, TX, Fort Worth, 761119"
}
```

--- 
### Sample meeting
Meeting: [Prairie Dog Park Public Meeting - Revised Master Plan](https://www.fortworthtexas.gov/departments/parks/events/meetings-and-projects/prairie-dog-revised-master-meeting) on December 14, 2024
```json
{
  "data": {
    "Title": "Prairie Dog Park Public Meeting - Revised Master Plan",
    "Description": "Fort Worth Park & Recreation invites you to participate in helping plan the future updates and/or upgrades to Prairie Dog Park.",
    "Link": "https://www.fortworthtexas.gov/departments/parks/events/meetings-and-projects/prairie-dog-revised-master-meeting",
    "Image": "/files/assets/public/v/1/parks-and-recreation/images/public-meetings/prairie-dog-rendering.jpg",
    "AltText": "prairie-dog-rendering.jpg",
    "Address": {
      "Venue": "Eugene McCray Community Center",
      "Street": "4932 Wilbarger St, Fort Worth, TX",
      "Suburb": "Fort Worth",
      "PostCode": "761119",
      "Formatted": "Eugene McCray Community Center, 4932 Wilbarger St, Fort Worth, TX, Fort Worth, 761119"
    },
    "AdditionalInfo": [],
    "IsCancelled": false
  },
  "success": true
}
```